fe/fe-core/src/main/java/com/starrocks/connector/metadata/MetadataTableName.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector.metadata;
+
+import com.starrocks.connector.exception.StarRocksConnectorException;
+
+import java.util.Locale;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MetadataTableName {
+    private static final Pattern TABLE_PATTERN = Pattern.compile("" +
```

Comment:
we  can not get table name and table type at parser?

Reply:
yes, our olap table table name can contain `$`